pkg/generate/git/list_remote_timeout_test.go

```diff
@@ -0,0 +1,47 @@
+package git
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+)
+
+// TestListRemoteTimeout tests that the `oc new-app` machinery that invokes `git ls-remote`
+// on uncooperative servers correctly times out
+func TestListRemoteTimeout(t *testing.T) {
+	timeout := 1 * time.Millisecond
```

Comment:
Since we are not using a `httptest.Server` anymore and simply running a command (currently `yes`), some of those issues are alleviated.